### PR TITLE
#13792 : Improving the DB connection handling in the Content Handler.

### PR DESCRIPTION
**Pull request by jcastro-dotcms**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publisher/receiver/BundlePublisher.java [hunk 1]
```diff
@@ -193,65 +193,58 @@ public PublisherConfig process ( final PublishStatus status ) throws DotPublishi
...
-                Logger.error(BundlePublisher.class, "Unable to update audit table : " + e.getMessage(), e);
+
+                auditAPI.updatePublishAuditStatus(bundleFolder, PublishAuditStatus.Status.FAILED_TO_PUBLISH,
+                        currentStatusHistory);
+            } catch (DotPublisherException e1) {
+                throw new DotPublishingException("Cannot update audit of bundle with ID '" + bundleName + "': ", e);
             }
+            throw new DotPublishingException("Error publishing bundle with ID '" + bundleName + "': " + e, e);
         } finally {
             DbConnectionFactory.closeSilently();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/receiver/BundlePublisher.java]: Keep in mind you only can close a connection that you open!

File: dotCMS/src/main/java/com/dotcms/api/system/event/SystemEventsFactory.java [hunk 2]
```diff
@@ -161,10 +161,8 @@ private void push(final SystemEvent systemEvent, boolean forceNewTransaction) th
 				Logger.error(this, msg, e);
 				throw new DotDataException(msg, e);
 			} finally {
-
-				if (localTransaction) {
-
-					DbConnectionFactory.closeSilently();
+				if (localTransaction && isNewConnection) {
+					HibernateUtil.closeSessionSilently();
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/api/system/event/SystemEventsFactory.java]: @jcastro-dotcms @jdotcms shouldn't we use the annotation approach here instead?

File: dotCMS/src/main/java/com/dotcms/publisher/receiver/BundlePublisher.java (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/publisher/receiver/BundlePublisher.java]: I just added more data to the error message, did not change the code.

File: dotCMS/src/main/java/com/dotcms/api/system/event/SystemEventsFactory.java (hunk 2 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/api/system/event/SystemEventsFactory.java]: This change was suggested directly by @jdotcms

Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/api/system/event/SystemEventsFactory.java]: The thing is there is some scenario where the transaction is force and it is not supported by the annotation

**Review by jgambarios — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**